includes/polling/ports.inc.php

```diff
@@ -470,16 +472,18 @@
             $this_port['ifPhysAddress']              = zeropad($a_a).zeropad($a_b).zeropad($a_c).zeropad($a_d).zeropad($a_e).zeropad($a_f);
         }
 
-        if (is_numeric($this_port['ifHCInBroadcastPkts']) && is_numeric($this_port['ifHCOutBroadcastPkts']) && is_numeric($this_port['ifHCInMulticastPkts']) && is_numeric($this_port['ifHCOutMulticastPkts']) && $device['os'] !== 'ciscosb') {
-            echo 'HC ';
-            $this_port['ifInBroadcastPkts']  = $this_port['ifHCInBroadcastPkts'];
-            $this_port['ifOutBroadcastPkts'] = $this_port['ifHCOutBroadcastPkts'];
-            $this_port['ifInMulticastPkts']  = $this_port['ifHCInMulticastPkts'];
-            $this_port['ifOutMulticastPkts'] = $this_port['ifHCOutMulticastPkts'];
+        // use HC values if they are available
+        foreach ($hc_mappings as $hc_oid => $if_oid) {
+            if (isset($this_port[$hc_oid]) && $this_port[$hc_oid]) {
```

Comment:
is_numeric() doesn't seem worthwhile.  `$this_port[$hc_oid] == true` (the == true is implied in the code) covers empty strings and 0 values. is_numeric() only covers empty strings.

Excluding ciscosb was because the hc oids are present, but at least some of them don't work and only return 0. When we check each stat, we can exclude 0 values because it only affects that individual stat.

Treating the HC stats as a group really bit us because of inconsistent vendor implementations.